misk/src/main/kotlin/misk/web/WebActionModule.kt

```diff
@@ -5,27 +5,78 @@ import misk.web.actions.WebAction
 import misk.web.actions.WebActionEntry
 import kotlin.reflect.KClass
 
-@Suppress("DEPRECATION")
 class WebActionModule<A : WebAction> private constructor(
-  val webActionClass: KClass<A>
+  val actionClass: KClass<A>,
+  val url_path_prefix: String
 ) : KAbstractModule() {
   override fun configure() {
-    multibind<WebActionEntry>().toInstance(WebActionEntry(webActionClass))
+    multibind<WebActionEntry>().toInstance(WebActionEntry(actionClass, url_path_prefix))
+    // Ensures that the action has an @Inject annotation and that its dependencies are satisfied
+    getProvider(actionClass.java)
   }
 
   companion object {
-    @Deprecated("use multibind<WebActionEntry>().toInstance(WebActionEntry(...))")
+    /**
+     * Registers a web action.
+     * @param actionClass: The web action to register.
+     */
     inline fun <reified A : WebAction> create(): WebActionModule<A> = create(A::class)
 
-    @Deprecated("use multibind<WebActionEntry>().toInstance(WebActionEntry(...))")
+    /**
+     * Registers a web action.
+     * @param actionClass: The web action to register.
+     */
     @JvmStatic
-    fun <A : WebAction> create(webActionClass: Class<A>): WebActionModule<A> {
-      return create(webActionClass.kotlin)
+    fun <A : WebAction> create(actionClass: Class<A>): WebActionModule<A> {
+      return create(actionClass.kotlin)
     }
 
-    @Deprecated("use multibind<WebActionEntry>().toInstance(WebActionEntry(...))")
-    fun <A : WebAction> create(webActionClass: KClass<A>): WebActionModule<A> {
-      return WebActionModule(webActionClass)
+    /**
+     * Registers a web action.
+     */
+    fun <A : WebAction> create(actionClass: KClass<A>): WebActionModule<A> {
+      return WebActionModule(actionClass, "/")
+    }
+
+    /**
+     * Registers a web action with a path prefix.
+     * @param url_path_prefix: defaults to "/". If not empty, must match pattern requirements:
```

Comment:
rather than copy-pasting this 3x can you link to a single definition?

Reply:
kotlindoc doesn't support linking to overloaded methods, but I'll just delete the redundant ones.